src/pubtools/_marketplacesvm/tasks/delete/command.py

```diff
@@ -0,0 +1,271 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import datetime
+import json
+import logging
+import os
+from typing import Any, Dict, Iterator, List
+
+from attrs import asdict, evolve
+from more_executors import Executors
+from pushsource import AmiPushItem, Source, VMIPushItem
+
+from ...arguments import SplitAndExtend
+from ...services import CloudService, CollectorService
+from ...services.rhsm import AwsRHSMClientService
+from ...task import RUN_RESULT, MarketplacesVMTask
+from ..push.items import State
+
+log = logging.getLogger("pubtools.marketplacesvm")
+
+
+class VMDelete(MarketplacesVMTask, CloudService, CollectorService, AwsRHSMClientService):
+    """Delete an image from a cloud provider."""
+
+    _REQUEST_THREADS = int(os.environ.get("COMBINED_DELETE_REQUEST_THREADS", "5"))
+    _SKIPPED = False
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the CommunityVMPush instance."""
+        super(VMDelete, self).__init__(*args, **kwargs)
+
+    @property
+    def raw_items(self) -> Iterator[AmiPushItem]:
+        """
+        Load all push items from the given source(s) and yield them.
+
+        Yields:
+            The AmiPushItems from the given sources.
+        """
+        for source_url in self.args.source:
+            with Source.get(source_url) as source:
+                log.info("Loading items from %s", source_url)
+                for item in source:
+                    if not isinstance(item, AmiPushItem):
+                        log.warning(
+                            "Push Item %s at %s is not an AmiPushItem, dropping it from the queue.",
+                            item.name,
+                            item.src,
+                        )
+                        continue
+                    self.builds_borg.received_builds.add(item.build_info.id)
+                    yield item
+
+    @property
+    def mapped_items(self) -> List[AmiPushItem]:  # type: ignore [override]
+        """
+        Return the mapped push item with destinations and metadata from StArMap.
+
+        Returns
+            The wrapped push item with the additional information from StArMap.
+        """
+        mapped_items = []
+        for item in self.raw_items:
+            mapped_items.append(item)
+        return mapped_items
```

Comment:
Since StArMap is not required in this step, maybe it would be simpler to just re-yield what the `raw_items` produce?

```suggestion
        for item in self.raw_items:
            yield item
```

Or completely remove it as we don't expect anything from StArMap, so you can directly read from `self.raw_items`

Reply:
I had this to make this easier:
max_workers=min(max(len(mapped_items), 1), self._REQUEST_THREADS),

Getting the length of a yielded item seemed to be not working. If you have suggestions let me know.

Reply:
What about using this directly inside `run`?

```
mapped_items = [x for x in self.raw_items]
```

Reply:
fixed

Reply:
tyvm